apollo-backend/src/test/java/io/logz/apollo/helpers/RealDeploymentGenerator.java

```diff
@@ -149,7 +149,10 @@ public Environment getEnvironment() {
         return environment;
     }
 
-    public void setEnvironment(Environment environment) { this.environment = environment; }
+    public void setEnvironment(Environment environment) {
+        this.environment = environment;
+        this.deployment.setEnvironmentId(environment.getId());
```

Comment:
Add this to setService as well

Reply:
There is no setService in RealDeploymentGenerator